File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> mp;
        mp[0]=1;
        int ans =0;
        int total = 0 ;
        for( int i : nums)
        {
             total +=i;
             if(mp.find(total-k)!=mp.end()) ans+=mp[total-k];
             mp[total]++;

        }
        return ans;
    }
};

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int sum = 0;
        int ans = 0;
        Map<Integer, Integer> mp = new HashMap<>();
        for(int i : nums)
        {
            sum+=i;
            if(sum==k) ans++;
            if(mp.get(sum-k)!=null) ans +=mp.get(sum-k);
            mp.put(sum,mp.getOrDefault(sum,0)+1);
        }
        return ans;
    }
}